### PR TITLE
update shadow firewall support

### PR DESCRIPTION
**Pull request by ericyz**
The goal of this PR is to add visibility of the traffic between the Google-managed default GKE firewall. The context of the request is that there are some compliance-driven requirements to capture all the inter-node traffics, which doesn't provide by the default GKE firewalls out of box.

The solution including in this PR can be described as:

- Create the same firewall to match the default firewall created by GKE
- Turn on the firewall logs for the created firewalls

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./firewall.tf:87:  Create shadow firewall rules to capture the 
./firewall.tf:110:  
./firewall.tf:128:  allow { 
./firewall.tf:130:    ports = ["10250", "443"] 
./firewall.tf:148:  allow { 
./firewall.tf:152:  allow { 
./firewall.tf:157:  allow { 
./modules/beta-public-cluster-update-variant/firewall.tf:88:  Create shadow firewall rules to capture the 
./modules/beta-public-cluster-update-variant/firewall.tf:111:  
./modules/beta-public-cluster-update-variant/firewall.tf:129:  allow { 
./modules/beta-public-cluster-update-variant/firewall.tf:131:    ports = ["10250", "443"] 
./modules/beta-public-cluster-update-variant/firewall.tf:149:  allow { 
./modules/beta-public-cluster-update-variant/firewall.tf:153:  allow { 
./modules/beta-public-cluster-update-variant/firewall.tf:158:  allow { 
./modules/beta-private-cluster-update-variant/firewall.tf:82:  Create shadow firewall rules to capture the 
./modules/beta-private-cluster-update-variant/firewall.tf:105:  
./modules/beta-private-cluster-update-variant/firewall.tf:123:  allow { 
./modules/beta-private-cluster-update-variant/firewall.tf:125:    ports = ["10250", "443"] 
./modules/beta-private-cluster-update-variant/firewall.tf:143:  allow { 
./modules/beta-private-cluster-update-variant/firewall.tf:147:  allow { 
./modules/beta-private-cluster-update-variant/firewall.tf:152:  allow { 
./modules/private-cluster-update-variant/firewall.tf:82:  Create shadow firewall rules to capture the 
./modules/private-cluster-update-variant/firewall.tf:105:  
./modules/private-cluster-update-variant/firewall.tf:123:  allow { 
./modules/private-cluster-update-variant/firewall.tf:125:    ports = ["10250", "443"] 
./modules/private-cluster-update-variant/firewall.tf:143:  allow { 
./modules/private-cluster-update-variant/firewall.tf:147:  allow { 
./modules/private-cluster-update-variant/firewall.tf:152:  allow { 
./modules/private-cluster/firewall.tf:82:  Create shadow firewall rules to capture the 
./modules/private-cluster/firewall.tf:105:  
./modules/private-cluster/firewall.tf:123:  allow { 
./modules/private-cluster/firewall.tf:125:    ports = ["10250", "443"] 
./modules/private-cluster/firewall.tf:143:  allow { 
./modules/private-cluster/firewall.tf:147:  allow { 
./modules/private-cluster/firewall.tf:152:  allow { 
./modules/beta-private-cluster/firewall.tf:82:  Create shadow firewall rules to capture the 
./modules/beta-private-cluster/firewall.tf:105:  
./modules/beta-private-cluster/firewall.tf:123:  allow { 
./modules/beta-private-cluster/firewall.tf:125:    ports = ["10250", "443"] 
./modules/beta-private-cluster/firewall.tf:143:  allow { 
./modules/beta-private-cluster/firewall.tf:147:  allow { 
./modules/beta-private-cluster/firewall.tf:152:  allow { 
./modules/beta-public-cluster/firewall.tf:88:  Create shadow firewall rules to capture the 
./modules/beta-public-cluster/firewall.tf:111:  
./modules/beta-public-cluster/firewall.tf:129:  allow { 
./modules/beta-public-cluster/firewall.tf:131:    ports = ["10250", "443"] 
./modules/beta-public-cluster/firewall.tf:149:  allow { 
./modules/beta-public-cluster/firewall.tf:153:  allow { 
./modules/beta-public-cluster/firewall.tf:158:  allow { 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
firewall.tf
--- old/firewall.tf
+++ new/firewall.tf
@@ -88,7 +88,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -100,16 +100,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -125,12 +125,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -145,18 +145,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/beta-public-cluster-update-variant/firewall.tf
--- old/modules/beta-public-cluster-update-variant/firewall.tf
+++ new/modules/beta-public-cluster-update-variant/firewall.tf
@@ -89,7 +89,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -101,16 +101,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -126,12 +126,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -146,18 +146,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/beta-private-cluster-update-variant/firewall.tf
--- old/modules/beta-private-cluster-update-variant/firewall.tf
+++ new/modules/beta-private-cluster-update-variant/firewall.tf
@@ -83,7 +83,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -95,16 +95,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -120,12 +120,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -140,18 +140,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/private-cluster-update-variant/firewall.tf
--- old/modules/private-cluster-update-variant/firewall.tf
+++ new/modules/private-cluster-update-variant/firewall.tf
@@ -83,7 +83,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -95,16 +95,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -120,12 +120,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -140,18 +140,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/private-cluster/firewall.tf
--- old/modules/private-cluster/firewall.tf
+++ new/modules/private-cluster/firewall.tf
@@ -83,7 +83,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -95,16 +95,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -120,12 +120,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -140,18 +140,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/beta-private-cluster/firewall.tf
--- old/modules/beta-private-cluster/firewall.tf
+++ new/modules/beta-private-cluster/firewall.tf
@@ -83,7 +83,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -95,16 +95,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -120,12 +120,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -140,18 +140,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/beta-public-cluster/firewall.tf
--- old/modules/beta-public-cluster/firewall.tf
+++ new/modules/beta-public-cluster/firewall.tf
@@ -89,7 +89,7 @@
   traffic flow between the managed firewall rules
  *****************************************/
 resource "google_compute_firewall" "shadow_allow_pods" {
-  count       = var.add_shadow_firewall_rules ? 1 : 0
+  count = var.add_shadow_firewall_rules ? 1 : 0
 
   name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
   description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
@@ -101,16 +101,16 @@
   source_ranges = [local.cluster_alias_ranges_cidr[var.ip_range_pods]]
   target_tags   = [local.cluster_network_tag]
 
- # Allow all possible protocols
+  # Allow all possible protocols
   allow { protocol = "tcp" }
   allow { protocol = "udp" }
   allow { protocol = "icmp" }
   allow { protocol = "sctp" }
   allow { protocol = "esp" }
   allow { protocol = "ah" }
-  
+
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -126,12 +126,12 @@
   source_ranges = [local.cluster_endpoint_for_nodes]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["10250", "443"] 
+    ports    = ["10250", "443"]
   }
   log_config {
-   metadata = "INCLUDE_ALL_METADATA"
+    metadata = "INCLUDE_ALL_METADATA"
   }
 }
 
@@ -146,18 +146,18 @@
   source_ranges = [local.cluster_subnet_cidr]
   target_tags   = [local.cluster_network_tag]
 
-  allow { 
+  allow {
     protocol = "icmp"
   }
 
-  allow { 
+  allow {
     protocol = "udp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
-  allow { 
+  allow {
     protocol = "tcp"
-    ports = ["1-65535"]
+    ports    = ["1-65535"]
   }
 
   log_config {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by morgante** [comment]
@ericyz Please rebase on master.

**Comment by ericyz** [comment]
Hey team, any feedback on this PR?

**Review by morgante — COMMENTED** [review]
@ericyz Can you explain more about the purpose of "shadow" firewall rules? Maybe we could have an explanation somewhere in the README or description?

**Comment by ericyz** [comment]
Hi @morgante , I have added more descriptions to the TF variables. The purpose of shadow firewall is to gain more network insight of the packages flowing between GKE worker and master nodes.

File: firewall.tf [hunk 1]
```diff
@@ -82,3 +82,89 @@ resource "google_compute_firewall" "master_webhooks" {
...
+
+/******************************************
+  Create shadow firewall rules to capture the
+  traffic flow between the managed firewall rules
+ *****************************************/
+resource "google_compute_firewall" "shadow_allow_pods" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-all"
+  description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow pod communication."
```
Comment (morgante) [inline firewall.tf:95]:
```suggestion
  description = "Managed by terraform gke module: A shadow firewall rule to match the default rule allowing pod communication."
```

File: firewall.tf [hunk 2]
```diff
@@ -82,3 +82,89 @@ resource "google_compute_firewall" "master_webhooks" {
...
+  log_config {
+    metadata = "INCLUDE_ALL_METADATA"
+  }
+}
+
+resource "google_compute_firewall" "shadow_allow_master" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-master"
+  description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow master and woker nodes communication."
```
Comment (morgante) [inline firewall.tf:121]: Typo:

```suggestion
  description = "Managed by terraform gke module: A shadow firewall rule to match the default rule allowing master and woker nodes communication."
```

File: firewall.tf [hunk 3]
```diff
@@ -82,3 +82,89 @@ resource "google_compute_firewall" "master_webhooks" {
...
+  log_config {
+    metadata = "INCLUDE_ALL_METADATA"
+  }
+}
+
+resource "google_compute_firewall" "shadow_allow_nodes" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-vms"
+  description = "Managed by terraform gke module: A shadow firewall rule to match the fireall allow woker nodes communication."
```
Comment (morgante) [inline firewall.tf:144]: Typo:

```suggestion
  description = "Managed by terraform gke module: A shadow firewall rule to match the default rule allowing worker node communication."
```

**Review by morgante — CHANGES_REQUESTED** [review]

File: autogen/main/firewall.tf.tmpl [hunk 4]
```diff
@@ -95,3 +95,89 @@ resource "google_compute_firewall" "master_webhooks" {
...
+  log_config {
+    metadata = "INCLUDE_ALL_METADATA"
+  }
+}
+
+resource "google_compute_firewall" "shadow_allow_master" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-master"
+  description = "Managed by terraform gke module: A shadow firewall rule to match the default rule allowing woker nodes communication."
```
Comment (morgante) [inline autogen/main/firewall.tf.tmpl:134]:
```suggestion
  description = "Managed by terraform GKE module: A shadow firewall rule to match the default rule allowing worker nodes communication."
```

File: autogen/main/firewall.tf.tmpl [hunk 5]
```diff
@@ -95,3 +95,89 @@ resource "google_compute_firewall" "master_webhooks" {
...
+  log_config {
+    metadata = "INCLUDE_ALL_METADATA"
+  }
+}
+
+resource "google_compute_firewall" "shadow_allow_nodes" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-vms"
+  description = "Managed by terraform gke module: A shadow firewall rule to match the default rule allowing woker nodes communication."
```
Comment (morgante) [inline autogen/main/firewall.tf.tmpl:157]:
```suggestion
  description = "Managed by Terraform GKE module: A shadow firewall rule to match the default rule allowing worker nodes communication."
```

**Review by morgante — CHANGES_REQUESTED** [review]

**Comment by ericyz** [comment]
Thanks @morgante for updating my type. Please review and let me know anything to update for the merge.

**Review by morgante — APPROVED** [review]

**Pull request merged**